### DISCUSSION TITLE
Skip Java dependency updated from JDK17 to JDK21

### DISCUSSION THREAD

**Discussion by marcprux** [skiptools/skip#94] [Announcements]
We have released Skip [0.8.10](https://github.com/skiptools/skip/releases/tag/0.8.10), along with skip-unit [0.6.0](https://github.com/skiptools/skip-unit/releases/tag/0.6.0), which updates the Skip Java dependency from JDK 17 to JDK 21 by way of skip-unit's `skip.yml` dependency specification. Java 21, which is a long-term support (LTS) release, has been the default for Homebrew for a few months now, but Gradle did not support it well until 8.6 was released.

This update should be transparent for users who update their Skip packages using Xcode's `File / Packages / Update to Latest Package Versions`. In fact, initial builds should be faster and more robust, due to the no longer needing to "auto-provision" (i.e., download) a separate JDK 17 install when it is not already installed.

Please do let us know if you encounter issues that seem related to the Java update. Happy Skipping!

**Comment by marcprux** [comment]
Some additional dependency updates:

skip: SwiftSyntax updated from 509.0.0 to 509.1.1
skip-model: androidx-compose-bom has been updated from 2023.10.01 to 2024.02.00

**Comment by pnewell** [comment]
I can't build anymore since upgrading. I've also tried making a fresh project and same issue when I try to build from Xcode. Oddly if I open the Android project directly in AndroidStudio it builds fine. 

```
openjdk version "21.0.1" 2023-10-17
```

I keep getting this error:

```
GRADLE> Starting a Gradle Daemon (subsequent builds will be faster)
GRADLE> 
GRADLE> FAILURE: Build failed with an exception.
GRADLE> 
GRADLE> * What went wrong:
GRADLE> Could not determine the dependencies of task ':skip-plugins:compileKotlin'.
GRADLE> > Unknown Kotlin JVM target: 21
GRADLE> 
GRADLE> * Try:
GRADLE> > Run with --stacktrace option to get the stack trace.
GRADLE> > Run with --info or --debug option to get more log output.
GRADLE> > Run with --scan to get full insights.
GRADLE> > Get more help at https://help.gradle.org.
GRADLE> 
GRADLE> BUILD FAILED in 25s
```

**Comment by marcprux** [comment]
Can you try manually setting the project to use JDK 20 or 21 in the Settings > Build, Execution, Deployment > Build Tools > Gradle setting section, as per this screenshot:

<img width="1148" alt="Screenshot 2024-02-24 at 19 24 53" src="https://github.com/skiptools/skip/assets/659086/dead6cb2-42b4-4995-8e96-acb83f403c94">

Please let us know if this works. Ideally, we will be able to set it for you – we will look into it.

For more details on this configuration setting, see https://developer.android.com/build/jdks#jdk-config-in-studio

**Comment by pnewell** [comment]
@marcprux I had already tried that and it still wasn't working, but for anyone else that gets stuck on this error, it was actually that I needed to update gradle, running `brew install gradle` resolved it for me
